addons/interactions/src/instrument.ts

```diff
@@ -0,0 +1,239 @@
+
+      // We need to throw to break out of the play function, but we don't want to trigger a redbox
+      // so we throw an ignoredException, which is caught and silently ignored by Storybook.
+      if (call.interceptable) {
+        throw IGNORED_EXCEPTION;
+      }
+
+      // Non-interceptable calls need their exceptions forwarded to the next interceptable call.
+      iframeState.forwardedException = e;
+      return e;
+    }
+    throw e;
+  }
+}
+
+function intercept(fn: Function, call: Call) {
+  // For a "jump to step" action, continue playing until we hit a call by that ID.
+  // For chained calls, we can only return a Promise for the last call in the chain.
+  const isChainedUpon = sharedState.chainedCallIds.has(call.id);
+  if (!sharedState.isDebugging || isChainedUpon || sharedState.playUntil) {
+    if (sharedState.playUntil === call.id) sharedState.playUntil = undefined;
+    return invoke(fn, call);
+  }
+
+  // Instead of invoking the function, defer the function call until we continue playing.
+  return new Promise((resolve) => {
+    iframeState.next[call.id] = resolve;
+  })
+    .then(() => delete iframeState.next[call.id])
+    .then(() => invoke(fn, call));
+}
+
+// Monkey patch an object method to record calls.
+// Returns a function that invokes the original function, records the invocation ("call") and
+// returns the original result.
+function track(method: string, fn: Function, args: any[], { path = [], ...options }: Options) {
+  const id = `${(iframeState.n += 1)}-${method}`;
+  const { intercept: interceptable = false, retain = false } = options;
+  const call: Call = { id, path, method, args, interceptable, retain };
+  const result = (options.intercept ? intercept : invoke)(fn, call);
+  return instrument(result, { ...options, mutate: true, path: [{ __callId__: call.id }] });
+}
+
+function patch(method: string, fn: Function, options: Options) {
+  const patched = (...args: any[]) => track(method, fn, args, options);
+  Object.defineProperty(patched, 'name', { value: method, writable: false });
+  return patched;
+}
+
+function initialize() {
+  try {
+    if (!global.window.__STORYBOOK_ADDON_TEST_PREVIEW__) {
+      global.window.__STORYBOOK_ADDON_TEST_PREVIEW__ = {
+        n: 0,
+        next: {},
+        callRefsByResult: new Map(),
+      };
+    }
+    if (!global.window.parent.__STORYBOOK_ADDON_TEST_MANAGER__) {
+      global.window.parent.__STORYBOOK_ADDON_TEST_MANAGER__ = {
+        isDebugging: false,
+        chainedCallIds: new Set<Call['id']>(),
+        playUntil: undefined,
+      };
+    }
+
+    channel = addons.getChannel();
+    iframeState = global.window.__STORYBOOK_ADDON_TEST_PREVIEW__;
+    sharedState = global.window.parent.__STORYBOOK_ADDON_TEST_MANAGER__;
+
+    channel.on(EVENTS.NEXT, () => Object.values(iframeState.next).forEach((resolve) => resolve()));
+    channel.on(EVENTS.RELOAD, () => global.window.location.reload());
```

Comment:
Are we getting rid of this in a later iteration? We talked about adding `Events.FORCE_RELOAD`, which would behave the same/opposite to `FORCE_RE_RENDER`

Reply:
Certainly this should be replaces with a Storybook-native FORCE_RELOAD event as discussed. Do you have suggestions on how/where to implement that?